src/Cardano/Shell/DaedalusIPC.hs

```diff
@@ -56,24 +55,25 @@ instance ToJSON MsgOut where
 -- If the IPC channel was successfully set up, this function won't return until
 -- the parent process exits. Otherwise, it will return immediately. Before
 -- returning, it will log an message about why it has exited.
+--
+-- TODO(KS): If you want to use TRACE here, you need to provide the trace functions
+-- as params OR provide a record of trace functions.
 daedalusIPC
-    :: Trace IO Text
-    -- ^ Logging object
-    -> Int
+    :: Int
     -- ^ Port number to send to Daedalus
     -> IO ()
-daedalusIPC tr port = setupNodeChannel >>= \case
+daedalusIPC port = setupNodeChannel >>= \case
     Right chan -> do
-        logInfo tr "Daedalus IPC server starting"
+        putTextLn "Daedalus IPC server starting"
```

Comment:
Perhaps it will make sense to use the `contratrace` `Tracer` abstraction here in future.

Reply:
The idea that I got was to do something like:
```
data LoggingFunctions = LoggingFunctions
    { logInfo :: () -> IO ()
    , ...
    }
```

And hide the implementation inside? We then don't even need that dependency if we want to push everything out.